Source/WebKit/UIProcess/API/Swift/WebPage+Transferable.swift

```diff
@@ -0,0 +1,257 @@
+
+#if ENABLE_SWIFTUI && compiler(>=6.0)
+
+public import CoreTransferable
+import Foundation
+import UniformTypeIdentifiers
+@_weakLinked internal import SwiftUI
+internal import WebKit_Private
+internal import WebKit_Private.WKSnapshotConfigurationPrivate
+
+@available(WK_IOS_TBA, WK_MAC_TBA, WK_XROS_TBA, *)
+@available(watchOS, unavailable)
+@available(tvOS, unavailable)
+extension WebPage: Transferable {
+    /// A specialized configuration of a specific exportable type that can have specific properties unique to the content type.
+    public nonisolated struct ExportedContentConfiguration: Sendable {
+        /// Represents a specific semantic region of a webpage.
+        public struct Region: Sendable {
+            private enum Storage {
+                case contents
+                case rect(CGRect)
+            }
+
+            /// A region that corresponds to a rectangle in the page's coordinate system.
+            ///
+            /// - Parameter rect: The rectangle to use for the region.
+            /// - Returns: A ``Region`` that uses the specified rectangle.
+            public static func rect(_ rect: CGRect) -> Self {
+                .init(storage: .rect(rect))
+            }
+
+            /// The entire region of the webpage.
+            public static var contents: Self {
+                .init(storage: .contents)
+            }
+
+            private let storage: Storage
+
+            private init(storage: Storage) {
+                self.storage = storage
+            }
+
+            fileprivate var rect: CGRect? {
+                switch storage {
+                case .contents: nil
+                case .rect(let value): value
+                }
+            }
+
+            fileprivate var usesContentsRect: Bool {
+                switch storage {
+                case .contents: true
+                case .rect(_): false
+                }
+            }
+        }
+
+        fileprivate enum Storage {
+            case pdf(Region, allowTransparentBackground: Bool)
+            case image(Region, allowTransparentBackground: Bool, snapshotWidth: CGFloat?, afterScreenUpdates: Bool)
+        }
+
+        /// A configuration of a webpage for a representation as PDF data.
+        ///
+        /// - Parameters:
+        ///   - region: The region of the page to generate the PDF for.
+        ///   - allowTransparentBackground: Indicates whether the PDF may have a transparent background.
+        /// - Returns: The PDF configuration of this page that will be used when producing its representation.
+        public static func pdf(region: Region = .contents, allowTransparentBackground: Bool = false) -> Self {
+            .init(storage: .pdf(region, allowTransparentBackground: allowTransparentBackground))
+        }
+
+        /// A configuration of a webpage for a representation as image data.
+        ///
+        /// - Parameters:
+        ///   - region: The region of the page to generate the image for.
```

Comment:
```suggestion
        ///   - region: The region of the page used to generate the image.
```

(thoughts? ending a sentence in `for` is a little informal)

Reply:
in_for_mal lol